ddl/sequence.go

```diff
@@ -0,0 +1,176 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"math"
+
+	"github.com/cznic/mathutil"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/ddl/util"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/meta"
+	math2 "github.com/pingcap/tidb/util/math"
+)
+
+func onCreateSequence(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	schemaID := job.SchemaID
+	tbInfo := &model.TableInfo{}
+	if err := job.DecodeArgs(tbInfo); err != nil {
+		// Invalid arguments, cancel this job.
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+
+	tbInfo.State = model.StateNone
+	err := checkTableNotExists(d, t, schemaID, tbInfo.Name.L)
+	if err != nil {
+		if infoschema.ErrDatabaseNotExists.Equal(err) || infoschema.ErrTableExists.Equal(err) {
+			job.State = model.JobStateCancelled
+		}
+		return ver, errors.Trace(err)
+	}
+
+	ver, err = updateSchemaVersion(t, job)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	switch tbInfo.State {
+	case model.StateNone:
+		// none -> public
+		tbInfo.State = model.StatePublic
+		tbInfo.UpdateTS = t.StartTS
+		err = createSequenceWithCheck(t, job, schemaID, tbInfo)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		// Finish this job.
+		job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
+		asyncNotifyEvent(d, &util.Event{Tp: model.ActionCreateSequence, TableInfo: tbInfo})
+		return ver, nil
+	default:
+		return ver, ErrInvalidDDLState.GenWithStackByArgs("sequence", tbInfo.State)
+	}
+}
+
+func createSequenceWithCheck(t *meta.Meta, job *model.Job, schemaID int64, tbInfo *model.TableInfo) error {
+	err := checkTableInfoValid(tbInfo)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return errors.Trace(err)
+	}
+	return t.CreateSequenceAndSetSeqValue(schemaID, tbInfo, tbInfo.Sequence.Start)
+}
+
+func handleSequenceOptions(SeqOptions []*ast.SequenceOption, sequenceInfo *model.SequenceInfo) {
+	var (
+		minSetFlag   bool
+		maxSetFlag   bool
+		startSetFlag bool
+	)
+	for _, op := range SeqOptions {
+		switch op.Tp {
+		case ast.SequenceOptionIncrementBy:
+			sequenceInfo.Increment = op.IntValue
+		case ast.SequenceStartWith:
+			sequenceInfo.Start = op.IntValue
+			startSetFlag = true
+		case ast.SequenceMinValue:
+			sequenceInfo.MinValue = op.IntValue
+			minSetFlag = true
+		case ast.SequenceMaxValue:
+			sequenceInfo.MaxValue = op.IntValue
+			maxSetFlag = true
+		case ast.SequenceCache:
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceNoCycle:
+			sequenceInfo.Cycle = false
+		}
+	}
+	// Fill the default value, min/max/start should be adjusted with increment's positive and negative.
```

Comment:
```suggestion
	// Fill the default value, min/max/start should be adjusted with the sign of sequenceInfo.Increment
```
"positive" and "negative" are both adjectives, they should not be used as objects.